### ISSUE TITLE
Support for ignoring raw data blocks after COPY ... FROM stdin;

### ISSUE THREAD

**Issue by Sanket17052006** [sqlfluff/sqlfluff#7697]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When parsing PostgreSQL dump files or fixture scripts that use `COPY ... FROM stdin;` followed by raw tab-separated data, `sqlfluff` attempts to parse the raw data as SQL. Because it is raw data and not standard SQL syntax, it immediately throws a `PRS` error.

### Expected Behaviour

When `sqlfluff` encounters `COPY [table] FROM stdin;`, it should ideally ignore all subsequent lines until it reaches the standard `\.` terminator, recognizing that the intervening lines are raw data and not SQL syntax to be linted.

### Observed Behaviour

`sqlfluff` throws a `PRS` (Found unparsable section) error at the very first line of the raw data.

```bash
$ sqlfluff lint --dialect postgres scripts/dev-instance/patron_data.sql
== [scripts/dev-instance/patron_data.sql] FAIL
L:  65 | P:   1 |  PRS | Line 65, Position 1: Found unparsable section:
                       | 'openlibrary\t15298516\t1\t\\N\t\\N\t2025-01-02 ...'

### How to reproduce

1. Create a file named ex- test1.sql
   ```sql
   COPY public.bookshelves_books (username, work_id, bookshelf_id) FROM stdin;
   openlibrary	15298516	1
   openlibrary	45310	3
   \.
   ```
2. Run the following command in your terminal:
   ```bash
   sqlfluff lint --dialect postgres test1.sql
   ```

### Dialect

postgres

### Version

4.1.0

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by RedZapdos123** [comment]
Reproduced with an MRE on sqlfluff 4.1.0.

#### Environment:
- Python 3.11.15
- sqlfluff 4.1.0

#### MRE SQL (test1.sql):
```
COPY public.bookshelves_books (username, work_id, bookshelf_id) FROM stdin;
openlibrary\t15298516\t1
openlibrary\t45310\t3
\\.
```

#### Reproduction Command:
sqlfluff lint --dialect postgres test1.sql

#### Observed:
- PRS parse error on the first raw data row after COPY ... FROM stdin.
- This confirms the stdin data block is being parsed as SQL instead of being ignored until the \\. terminator.
- CP01 keyword casing can also appear depending on rules/config, but the parser bug is the PRS on raw data lines.

Fixed in PR #7759.

#### The issue reproduced by the MRE:

<img width="1331" height="500" alt="Image" src="https://github.com/user-attachments/assets/c6b40070-1925-4489-b346-598e88c3fa1a" />
